### PR TITLE
feat: added follow/unfollow and bookmark

### PR DESCRIPTION
**Pull request by poornima-alugubelly**
- User can now follow/unfollow other users
- User can sort his feed by trending and date
- User can checkout explore feed
- User can now bookmark posts and view them

### CODE REVIEW

File: src/features/Posts/postSlice.js [hunk 1]
```diff
@@ -81,15 +86,63 @@ export const dislikePost = createAsyncThunk(
...
 );
 
+export const getBookmarks = createAsyncThunk(
+	"posts/getBookmarks",
+	async ({ token }) => {
+		try {
+			const response = await getAllBookmarkService(token);
+			return response.data;
+		} catch (error) {
+			console.log(error.response.data);
```
Comment (manojasarsa) [inline src/features/Posts/postSlice.js:96]: you can use `console.error` here

File: src/features/UserProfile/UserProfile.jsx [hunk 2]
```diff
@@ -1,74 +1,55 @@
...
+					key={post.id}
 				>
-					<SuggestedProfiles />
-				</GridItem>
-			</Grid>
-		</Box>
+					<Center>
+						<PostCard postDetails={post} key={post.id} />
+					</Center>
+					{/* <Divider orientation="horizontal" color={colorToggler(400)} /> */}
```
Comment (manojasarsa) [inline src/features/UserProfile/UserProfile.jsx:60]: you can remove this comment if no longer needed

**Review by manojasarsa — APPROVED** [review]
The code is very well structured and self-explantory.
Also, UI looks good.
GTG!

**Review by romabulani — APPROVED** [review]
- Clean and modular code
- Functionality is working as expected
- Good to merge
- Suggestion : After bookmarking the post, if I go to bookmarks and like the post from there, its not working, check that once. Rest all looks good.

File: src/features/Home/components/SortPosts.jsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+					<MenuButton as={Button} rightIcon={<BsChevronDown />}>
+						<Text> Date Posted</Text>
+					</MenuButton>
+					<MenuList bgColor={colorToggler(800)}>
+						<MenuItem
+							onClick={() => dispatch(sortByValue("oldest"))}
+							_hover={{ bgColor: colorToggler(900) }}
+						>
+							Oldest
+						</MenuItem>
```
Comment (himadri2110) [inline src/features/Home/components/SortPosts.jsx:51]: You can show the active state of the selected option from Trending, sort by oldest, sort by newest. 

Also, the 'Date Posted' modal always shows "oldest" as the selected option even if select "newest"

File: src/helpers/checkUserPresent.js [hunk 4]
```diff
@@ -1,4 +1,7 @@
-export const checkUserPresent = (userId, givenArray) => {
-	const res = givenArray.find((item) => item.id === userId);
+export const checkUserPresent = (givenUsername, givenArray) => {
+	const res = givenArray.find((item) => {
+		return item.username === givenUsername;
+	});
```
Comment (himadri2110) [inline src/helpers/checkUserPresent.js:4]: Nice way of making one function that can handle multiple cases 👍🏻

**Review by himadri2110 — APPROVED** [review]
The result of the edit and delete post functionalities isn't reflected immediately on the profile page. We need to go to the home page to see the updated result. 

Rest is all good.
Great work.!

**Pull request merged**